### ISSUE TITLE
Comment Content Block

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#30574]
Here is an initial take on the design for the Comment Content block. This block should only be available in the Site Editor, and potentially only when editing certain templates (Index, Single, Singular, Page).

![comment-content](https://user-images.githubusercontent.com/846565/113897418-bfaa7100-97c2-11eb-91fb-9b1211751bdf.png)

Design feedback is politely requested for all aspects:

* Icon
* Placeholder when there is no context present
* Toolbar actions
* Inspector settings
* Name
* Description

Figma link [here](https://www.figma.com/file/AagoBClTm0AgpfYCSKQl0m/Comment-blocks?node-id=501%3A1241).

**Comment by paaljoachim** [comment]
Hey James

Here I am wondering if we should have color controls (text and background). Perhaps even typography controls.
The icon looks good.

"The content of the comment." Add the punctuation. 

An additional control might be to add border control. As well as some padding.

**Comment by jameskoster** [comment]
Excellent feedback. I agree on all points :) 

If possible, it would be cool to use the design tools from https://github.com/WordPress/gutenberg/issues/27331 for the color, padding, and typography controls. But I don't know if they are ready yet.

**Comment by paaljoachim** [comment]
At the moment we can focus on what is ready. Taking one step at a time.
When additional controls are ready then these new controls can be added.

Of course the mockup can show controls that are not ready to be implemented, but when a developer picks up this issue and creates the block then he/she will see which controls are ready to be added.

**Comment by SantosGuillamot** [comment]
Hello! 👋   I would like to wrap up all the issues related to the Post Comments block to start working on a Comment Loop block as explained in [this tracking issue](https://github.com/WordPress/gutenberg/issues/34994). I'd like to make a quick summary to ensure we are aligned on this one:

## Description

Displays the content of the comment.

## Potential Settings

- **Typography**: It should be the same as the [Paragraph block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/paragraph).
- **Color Settings**: It should be the same as the [Paragraph block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/paragraph).
- **Layout Settings:** It should be the same as the [Post Content block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-content). It can inherit the default layout or change the width and content.

## References

- This should be similar to the [Post Content Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-content).
- There is already a basic version of the Post Comment Content Block: [GitHub code](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-content).

Of course, any feedback is welcome 🙂

**Comment by DAreRodz** [comment]
Hi all, :wave: I would like to create a draft PR and start working on this. @SantosGuillamot could you assign this issue to me? :blush:

**Comment by DAreRodz** [comment]
Hey, :wave: I had a few questions when I started working on this.

Looking at the design @jameskoster shared, it looks like the block shows a generic message when no context is provided. I understand that this happens when the comment ID is not specified in the parent block, am I right? What should the block show when the comment ID is not correct? Is it OK showing the same message?

Regarding the icon, I guess I have to create a new one inside `@wordpress/icons` using the design, right?

My last question is related to the settings this block will have. I've added those that @SantosGuillamot proposed, although in the design also appear some buttons to make the text bold o italic. I'm not sure, but these settings don't seem quite possible to be added as that would modify the comment content, if I'm not mistaken.

I'll keep working on adding tests and comments in the meantime. Thanks in advance. :slightly_smiling_face:
